### ISSUE TITLE
Allow transforming files while copying

### ISSUE THREAD

**Issue by tivac** [tivac/rollup-plugin-globsync#8]
For instance, when copying a `*.svg` file run it through [svgo](https://github.com/svg/svgo). Would need to be able to be `async` for obvious reasons.

There's already a `options.transform` that does file path transforms, feels like that should become `options.destination` and then `options.transform` becomes the `async ()=> {}` to optionally transform files.

Probably gets the original, un-transformed filepath as the only arg, any truthy response value will be used as the value and written to the ultimate destination instead of the original file value.

**Comment by panoply** [comment]
Yeah, it would be great if the transform option would allow modifications to passed in files, before copying/writing them to dest.

**Comment by panoply** [comment]
^ Support for transform as file modifications. No tests as of yet (thug life) but should do the job. If you like it, I will PR.
